### PR TITLE
Merging of Q# Language and Quantum Development Techniques into one single folder

### PR DESCRIPTION
**Pull request by geduardo**
Some comments on the PR:
-I don't have access to the Preview mode yet and since it is a big change probably there will be a lot of errors.
-I modified the toc.yml files manually. I don't know if it is the standard way of doing it.
-I'm not sure of how the cross-reference links work, so please check if what I did is correct.
-There is still some content overlapping and the organization should be improved, but I think it works as a first attempt.
-Summary of what I did:
1. I deleted the folders  "Q# programming language" and "Quantum development techniques". 
2. I created a new folder "Q# manual" with the content of "Q# programming language" and "Quantum development techniques" combined plus an introduction. As I'm not very experienced with Q# I tried to rewrite the minimum possible and respect most of the original writing in the merge. However, I wrote an introduction to the manual, added some content in some sections (e.g. in "The basics of Q#") and eliminated some overlapping content from both files, please check that out. 

I can provide a more detailed summary of all the changes I committed, just ask for it if needed. 

It's my first time so I hope I didn`t mess it up too much.

### CODE REVIEW

File: articles/manual/index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+and functions.
+
+Q# is scalable: it can be used to write simple demonstration programs like teleport that run on a few qubits, but also supports writing large, sophisticated programs such as simulations of complex molecules that will require large machines with millions of qubits. Even though large physical machines are still in the future, Q# allows a programmer to program complex quantum algorithms now. What is more, Q# supports various tasks such as debugging, profiling, resource estimation, and certain special-purpose simulations in a scalable way.
+
+
+# Why Q#?
+
+[![](http://img.youtube.com/vi/cOUrzxyng04/0.jpg)](http://www.youtube.com/watch?v=cOUrzxyng04 "")
+
+ While quantum hardware is still in an early developing stage, we need
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
 While quantum hardware is still in early development, we need 
```

File: articles/manual/index.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Q# is scalable: it can be used to write simple demonstration programs like teleport that run on a few qubits, but also supports writing large, sophisticated programs such as simulations of complex molecules that will require large machines with millions of qubits. Even though large physical machines are still in the future, Q# allows a programmer to program complex quantum algorithms now. What is more, Q# supports various tasks such as debugging, profiling, resource estimation, and certain special-purpose simulations in a scalable way.
+
+
+# Why Q#?
+
+[![](http://img.youtube.com/vi/cOUrzxyng04/0.jpg)](http://www.youtube.com/watch?v=cOUrzxyng04 "")
+
+ While quantum hardware is still in an early developing stage, we need 
+ to build adequate software for quantum computers to take advance
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
 to build adequate software for quantum computers to take advantage
```

File: articles/manual/index.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+Q# is scalable: it can be used to write simple demonstration programs like teleport that run on a few qubits, but also supports writing large, sophisticated programs such as simulations of complex molecules that will require large machines with millions of qubits. Even though large physical machines are still in the future, Q# allows a programmer to program complex quantum algorithms now. What is more, Q# supports various tasks such as debugging, profiling, resource estimation, and certain special-purpose simulations in a scalable way.
+
+
+# Why Q#?
+
+[![](http://img.youtube.com/vi/cOUrzxyng04/0.jpg)](http://www.youtube.com/watch?v=cOUrzxyng04 "")
+
+ While quantum hardware is still in an early developing stage, we need 
+ to build adequate software for quantum computers to take advance 
+ of the quantum power of computation when time comes. Q# is a tool
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
 of the quantum power of computation when the time comes. Q# is a tool 
```

File: articles/manual/index.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+# Why Q#?
+
+[![](http://img.youtube.com/vi/cOUrzxyng04/0.jpg)](http://www.youtube.com/watch?v=cOUrzxyng04 "")
+
+ While quantum hardware is still in an early developing stage, we need 
+ to build adequate software for quantum computers to take advance 
+ of the quantum power of computation when time comes. Q# is a tool 
+ made to provide a platform where developers can write and test programs 
+ to be run in a future generation of quantum computers.
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
 to be run on a future generation of quantum computers.
```

File: articles/manual/index.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+express quantum states. Some experience with complex numbers is crucial to
+understand the basics of quantum computation.
+* **Linear algebra (vectors and matrices):** quantum states are vectors 
+  in a vector space over the complex numbers. Quantum computing makes 
+  intense use of linear operators acting on these states. Therefore 
+  some level of expertise in linear algebra is indispensable to use Q#.
+* **Dirac notation:** most quantum computing texts use Dirac notation 
+  to express the mathematical concepts. Dirac notation is just a simple 
+  notation for linear algebra widely used in quantum physics.
+* **Basic concepts of quantum mechanics:** wether it is not required to
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
* **Basic concepts of quantum mechanics:** while it is not required to 
```

File: articles/manual/index.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+* **Linear algebra (vectors and matrices):** quantum states are vectors 
+  in a vector space over the complex numbers. Quantum computing makes 
+  intense use of linear operators acting on these states. Therefore 
+  some level of expertise in linear algebra is indispensable to use Q#.
+* **Dirac notation:** most quantum computing texts use Dirac notation 
+  to express the mathematical concepts. Dirac notation is just a simple 
+  notation for linear algebra widely used in quantum physics.
+* **Basic concepts of quantum mechanics:** wether it is not required to 
+  have a deep understanding of quantum mechanics to use Q#, it is highly 
+  recommendable to be familiar with some basic concepts, namely
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
  recommended to be familiar with some basic concepts, namely 
```

File: articles/manual/index.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+* **Dirac notation:** most quantum computing texts use Dirac notation 
+  to express the mathematical concepts. Dirac notation is just a simple 
+  notation for linear algebra widely used in quantum physics.
+* **Basic concepts of quantum mechanics:** wether it is not required to 
+  have a deep understanding of quantum mechanics to use Q#, it is highly 
+  recommendable to be familiar with some basic concepts, namely 
+  *quantum state*, *qubit*, *observable*, *measurement*, *Hamiltonian*,
+  *superposition*, *entanglement*, *interference*, among others.  
+## Learning materials
+Quantum computing is an emerging field and fortunately, there is an
```
Comment (bromeg) [inline articles/manual/index.md]:
```suggestion
Quantum computing is an emerging field and fortunately, there are an 
```

File: articles/manual/operations-functions.md [hunk 8]
```diff
@@ -283,3 +244,119 @@ function SquareOperation(op : (Qubit => Unit)) : (Qubit => Unit) {
...
 This approach will be used throughout the Q# standard library for expressing classical control flow in a way that can be readily used within quantum programs.
+
+# Generic Operations and Functions
+
+> [!TIP]
+> This section assumes some basic familiarity with [generics in C#](https://docs.microsoft.com/dotnet/csharp/programming-guide/generics/introduction-to-generics), [generics in F#](https://docs.microsoft.com/dotnet/fsharp/language-reference/generics/), [C++ templates](https://docs.microsoft.com/cpp/cpp/templates-cpp), or similar approaches to metaprogramming in other languages.
+
+Many functions and operations that we might wish to define do not actually heavily rely on the types of their inputs, but rather only implicitly use their types via some other function or operation.
+For example, consider the *map* concept common to many functional languages; given a function $f(x)$ and a collection of values $\{x_1, x_2, \dots, x_n\}$, map returns a new collection $\{f(x_1), f(x_2), \dots, f(x_n)\}$.
+To implement this in Q#, we can take advantage of that functions are first class.
```
Comment (bromeg) [inline articles/manual/operations-functions.md]:
```suggestion
To implement this in Q#, we can take advantage of first class functions.
```

File: articles/manual/operations-functions.md [hunk 9]
```diff
@@ -283,3 +244,119 @@ function SquareOperation(op : (Qubit => Unit)) : (Qubit => Unit) {
...
+    return mappedValues;
+}
+```
+
+In principle, we could write a version of `Map` for every pair of types that we encounter, but this introduces a number of difficulties.
+For instance, if we find a bug in `Map`, then we must ensure that the fix is applied uniformly across all versions of `Map`.
+Moreover, if we construct a new tuple or UDT, then we must now also construct a new `Map` to go along with the new type.
+While this is tractable for a small number of such functions, as we collect more and more functions of the same form as `Map`, the cost of introducing new types becomes unreasonably large in fairly short order.
+
+Much of this difficulty results, however, from that the we have not given the compiler the information it needs to recognize how the different versions of `Map` are related.
```
Comment (bromeg) [inline articles/manual/operations-functions.md]:
```suggestion
Much of this difficulty results, from not having given the compiler the information it needs to recognize how the different versions of `Map` are related.
```

File: articles/manual/index.md (hunk 7 above)
Comment (geduardo) [reply articles/manual/index.md]: In this one I think "is" is correct because "amount" is singular.

**Pull request closed without merging**